crates/holochain/src/core/ribosome/host_fn/must_get_valid_record.rs

```diff
@@ -117,3 +131,87 @@ pub fn must_get_valid_record(
     tracing::debug!(?ret);
     ret
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::conductor::api::error::ConductorApiError;
+    use crate::conductor::CellError;
+    use crate::core::ribosome::RibosomeError;
+    use crate::core::workflow::WorkflowError;
+    use crate::test_utils::RibosomeTestFixture;
+    use ::fixt::prelude::*;
+    use assert2::let_assert;
+    use holochain_serialized_bytes::prelude::*;
+    use holochain_state::prelude::{insert_op_cache, set_validation_status, set_when_integrated};
+    use holochain_timestamp::Timestamp;
+    use holochain_types::dht_op::{ChainOp, DhtOp, DhtOpHashed};
+    use holochain_wasm_test_utils::TestWasm;
+    use holochain_wasmer_host::prelude::*;
+    use holochain_zome_types::validate::ValidationStatus;
+
+    // This test ensures the ValidationStatus::Rejected arm is hit and returns a
+    // HostShortCircuit carrying ValidateCallbackResult::Invalid with the expected message.
+    #[tokio::test(flavor = "multi_thread")]
+    async fn must_get_valid_record_short_circuit_when_invalid_record_found() {
```

Comment:
This tests that `must_get_valid_record` returns an `ValidateCallbackResult::Invalid` when an invalid record is found.